src/components/button.ts

```diff
@@ -27,9 +34,10 @@ export interface ButtonProps {
   disabled?: boolean;
   primary?: boolean;
   border?: boolean;
-  status?: 'primary' | 'info' | 'success' | 'warning' | 'error';
-  additionalEvents?: Record<string, (event?: any) => any>;
+  status?: 'main' | 'primary' | 'info' | 'success' | 'warning' | 'error' | 'clear';
```

Comment:
`main` vs `primary` does not seem to make much sense in general status hierarchy, bit ambiguous?

Reply:
Definitely yes, but it will be a task for a major version change to v5.0.0. We'll have types for the buttons, and status will be replaced with `status-info`, `status-primary` etc. However we don't need to make a major and backwards incompatible change for the consumers.

Reply:
But what is main here? The gradient?

Reply:
Yes it is the gradient one. 
In v5 we'll arrange them under `type` and will be like:
```
type: 'main' | 'primary' | 'secondary' | 'clear';
// if type is "secondary"
status: Status;
```

we'll define them later

Reply:
Does a type of 'gradient' not make more sense then? 'main' and 'primary' are literal synonyms.

Reply:
Not for this case, since it is gradient now, but maybe in the future we'll replace it with something else. So it is same with naming with colors. There cannot be a `blue` or `red` button for example, since our lib is themeable and those can change.